src/mappings/calls/orderBook.ts

```diff
@@ -0,0 +1,64 @@
+import { SubstrateExtrinsic } from "@subql/types";
+import { addDataToHistoryElement, createHistoryElement, updateHistoryElementStats } from "../../utils/history";
+import { getAssetId, formatU128ToBalance } from '../../utils/assets';
+import { logStartProcessingCall } from "../../utils/logs";
+
+export async function orderBookPlaceLimitOrderHandler(extrinsic: SubstrateExtrinsic): Promise<void> {
+  logStartProcessingCall(extrinsic);
+
+  const historyElement = await createHistoryElement(extrinsic)
+
+  const { extrinsic: { args: [orderBookId, price, amount, side, lifespanOption] } } = extrinsic as any;
+
+  const base = getAssetId(orderBookId.base);
+  const quote = getAssetId(orderBookId.quote);
+  const details = {
+    dexId: orderBookId.dexId.toNumber(),
+    base,
+    quote,
+    orderId: null,
+    price: formatU128ToBalance(price.toString(), quote),
+    amount: formatU128ToBalance(amount.toString(), base),
+    side: side.toHuman(),
+    lifespan: !lifespanOption.isEmpty ? Number(lifespanOption.unwrap()) : null,
+  };
+
+  const limitOrderPlacedEvent = extrinsic.events.find(e =>
+    e.event.section === 'orderBook' &&
+    e.event.method === 'LimitOrderPlaced'
+  );
+
+  if (limitOrderPlacedEvent) {
+    const { event: { data: [orderBookId, orderId] } } = limitOrderPlacedEvent;
+    details.orderId = Number(orderId.toHuman());
+  }
+
+  await addDataToHistoryElement(extrinsic, historyElement, details)
+  await updateHistoryElementStats(extrinsic, historyElement);
+}
+
+export async function orderBookCancelLimitOrderHandler(extrinsic: SubstrateExtrinsic): Promise<void> {
+  logStartProcessingCall(extrinsic);
+
+  const historyElement = await createHistoryElement(extrinsic)
+  const cancelEvents = extrinsic.events.filter(e =>
+    e.event.section === 'orderBook' &&
+    e.event.method === 'LimitOrderCanceled'
```

Comment:
Does  method have the same name for batch and single?

Reply:
Yes, when order is canceled, backend generates the `orderBook.LimitOrderCancelled` event, no difference which extrinsic was called